src/include/function/comparison/vector_comparison_functions.h

```diff
@@ -13,17 +13,10 @@ struct ComparisonFunction {
     template<typename OP>
     static function_set getFunctionSet(const std::string& name) {
         function_set functionSet;
-        for (auto& comparableType : common::LogicalTypeUtils::getAllValidComparableLogicalTypes()) {
+        // We can only check whether two internal ids are equal or not.
+        for (auto& comparableType : common::LogicalTypeUtils::getAllValidLogicTypes()) {
```

Comment:
If we switch to all logical types, can u test the following

- compare between `RDF_VARAINT`
- compare between `INTERNAL_ID` especially for non equality comparison